dump of a fixed array of option strings fails to compile

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.1 8f69bd1.b5f71df
OS: linux, Ubuntu 20.04.6 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /space/v/vnew
vexe: /space/v/vnew/v
vexe mtime: 2023-09-29 02:48:24

vroot: OK, value: /space/v/vnew
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.42.0
Git vroot status: weekly.2023.39-19-g1476daf8
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg a.v`
```v
a := [?string('foo'), ?string(none)]!
dump(a)

```

**What did you expect to see?**

compiled program that runs

**What did you see instead?**
```
==================
/tmp/v_1000/a.946870798248241139.tmp.c:2074: error: cannot convert 'struct _option_string' to 'unsigned char *'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```
> [!NOTE]
> You can vote for this issue using the 👍 reaction. More votes increase the issue's priority for developers.
>
> Take into account that only the 👍 reaction counts as a vote.
> Only reactions to the issue itself will be counted as votes, not comments.

--- a/vlib/v/gen/c/auto_str_methods.v
+++ b/vlib/v/gen/c/auto_str_methods.v
@@ -678,27 +678,27 @@ fn (mut g Gen) gen_str_for_array_fixed(info ast.ArrayFixed, styp string, str_fn_
 				g.auto_str_funcs.writeln('\t\tif ( 0 == a[i] ) {')
 				g.auto_str_funcs.writeln('\t\t\tstrings__Builder_write_string(&sb, _SLIT("0"));')
 				g.auto_str_funcs.writeln('\t\t}else{')
-				g.auto_str_funcs.writeln('\t\t\tstrings__Builder_write_string(&sb, ${elem_str_fn_name}( ${deref} a[i]) );')
+				g.auto_str_funcs.writeln('\t\t\tstrings__Builder_write_string(&sb, ${elem_str_fn_name}(${deref}a[i]));')
 				g.auto_str_funcs.writeln('\t\t}')
 			} else {
-				g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${elem_str_fn_name}( ${deref} a[i]) );')
+				g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${elem_str_fn_name}(${deref}a[i]));')
 			}
-		} else if sym.kind in [.f32, .f64] {
+		} else if sym.kind in [.f32, .f64] && !typ.has_flag(.option) {
 			if sym.kind == .f32 {
 				field_str := str_intp_g32('a[i]')
-				g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${field_str} );')
+				g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${field_str});')
 			} else {
 				field_str := str_intp_g64('a[i]')
-				g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${field_str} );')
+				g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${field_str});')
 			}
-		} else if sym.kind == .string {
+		} else if sym.kind == .string && !typ.has_flag(.option) {
 			field_str := str_intp_sq('a[i]')
 			g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${field_str});')
-		} else if sym.kind == .rune {
-			tmp_str := str_intp_rune('${elem_str_fn_name}(  ${deref} a[i])')
+		} else if sym.kind == .rune && !typ.has_flag(.option) {
+			tmp_str := str_intp_rune('${elem_str_fn_name}(${deref}a[i])')
 			g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${tmp_str});')
 		} else {
-			g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${elem_str_fn_name}( ${deref} a[i]));')
+			g.auto_str_funcs.writeln('\t\tstrings__Builder_write_string(&sb, ${elem_str_fn_name}(${deref}a[i]));')
 		}
 	}
 	g.auto_str_funcs.writeln('\t\tif (i < ${info.size - 1}) {')

--- a/vlib/v/slow_tests/inout/dump_fixed_array_of_option.out
+++ b/vlib/v/slow_tests/inout/dump_fixed_array_of_option.out
@@ -1,0 +1,4 @@
+[vlib/v/slow_tests/inout/dump_fixed_array_of_option.vv:3] a1: [Option('foo'), Option(none)]
+[vlib/v/slow_tests/inout/dump_fixed_array_of_option.vv:6] a2: [Option(22), Option(none)]
+[vlib/v/slow_tests/inout/dump_fixed_array_of_option.vv:9] a3: [Option(22.2), Option(none)]
+[vlib/v/slow_tests/inout/dump_fixed_array_of_option.vv:12] a4: [Option(true), Option(none)]

--- a/vlib/v/slow_tests/inout/dump_fixed_array_of_option.vv
+++ b/vlib/v/slow_tests/inout/dump_fixed_array_of_option.vv
@@ -1,0 +1,13 @@
+fn main() {
+	a1 := [?string('foo'), ?string(none)]!
+	dump(a1)
+
+	a2 := [?int(22), ?int(none)]!
+	dump(a2)
+
+	a3 := [?f64(22.2), ?f64(none)]!
+	dump(a3)
+
+	a4 := [?bool(true), ?bool(none)]!
+	dump(a4)
+}

